File: problems/helpers/linkedlist/ListNode.py
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

def createLinkedList(li):
    assert len(li) > 0, "List is empty"
    root = ListNode(li[0])
    next = root
    for item in li[1:]:
        next.next = ListNode(item)
        next = next.next

    return root

def printLinkedList(node):
    if node != None:
        print(node.val, end=' ')
        printLinkedList(node.next)
    else:
        print()

File: problems/helpers/linkedlist/ListNode.go
package main
import "fmt"

type ListNode struct {
    Val int
    Next *ListNode
}

func createLinkedList(li []int) *ListNode {
	root := &ListNode{li[0], nil}
	next := root
	for _, item := range li[1:] {
		next.Next = &ListNode{item, nil}
		next = next.Next
	}
	return root
}

func printLinkedList(node *ListNode) {
	if node != nil {
		fmt.Print(node.Val, " ")
		printLinkedList(node.Next)
	} else {
		fmt.Println()
	}
}